nginx/config.go

```diff
@@ -73,8 +73,9 @@ http {
 	# treated as if it were HTTPs.
 	map $http_x_forwarded_proto $tmp_access_scheme {
 		default $scheme;               # if X-Forwarded-Proto header is empty, $tmp_access_scheme will be the actual protocol used
-		"~^(.*, ?)?http$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value
-		"~^(.*, ?)?https$" "https";    # account for the possibility of a comma-delimited X-Forwarded-Proto header value
+		"~^(.*, ?)?https(,.* ?)?$" "https";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if https is present
+		"~^(.*, ?)?http(,.* ?)?$" "http";      # account for the possibility of a comma-delimited X-Forwarded-Proto header value to see if http is present
+		"~^[A-Za-z][A-Za-z0-9+.-]*" $http_x_forwarded_proto; # Use the value passed in the header if its a valid URI scheme
```

Comment:
@kmala ok. Didn't know the regex was based off an RFC. If that regex describes what the RFC considers valid, cool, but I guess my question is more about validity with respect to what touter can actually handle. What happens, for instance, when the scheme is `xyz`?

Re: `ws` vs `wss`-- thank you for explaining that-- I didn't even know wss was a thing. I really _do_ need to research web sockets more. Do we need to specially account for these schemes in the way we do http and https so we can correctly determine the scenarios where we need to enforce SSL by redirecting?